Add support for repositoryName named equally to a repositoryUser

FrontSide opened the pull request:
If an image has the same name as a top level repository, the user got thrown back to the landing page and was thus not able to see the tags of this image. This fixes this issue for me.

For example:
A image stored on the registry at:
`private.registry.com:5000/redis`

is displayed on the frontend as follows:
![Redis Image](http://i.imgur.com/KRlA5fM.png)

Clicking this link currently doesn't show the user the image's tags but instead throws the user back to the landing page.

Commit:
if an image has the same name as a top level repository, the user got thrown back to the landing page and was thus not able to see the tags of this image. This commit fixes this issue.

FrontSide committed:
add comment

Morriz commented:
If this works and has no side effects, can it be merged? I am maintaining a fork based on this fix, but don't wish to just for that ;)

Commit:
Merge branch 'v2' of github.com:kwk/docker-registry-frontend into v2

Conflicts:
	app/app.js

Commit:
Add curl frontend container

Commit:
Add support for repoName==repoUser after update

After the tag detail change has been merged in, the changes made
before to support accessing repositories that have tha same name as user
are not sufficeient anymore.

With these chages it is now possible to list the detail tag page of all
repositories.

Commit:
Add readable host var

This variable will be used to display the hostname of the docker registry
on the tag detail page for the docker pull command.

Since as actual host, usually, the docker container 'registry'
is defined, 'registry' would also show up in this command
even though it's not the actual URL to the docker-registry
from which the image can be pulled.

FrontSide commented:
Not up to date anymore. Close.
My fork, however, supports this feature (also after the details addition).
